File: IDATT2106 Systemutvikling 2 med smidig prosjekt/backend/src/main/java/com/ntnu/gidd/service/token/RefreshTokenService.java
package com.ntnu.gidd.service.token;

import com.ntnu.gidd.model.RefreshToken;
import com.ntnu.gidd.security.token.JwtRefreshToken;
import com.ntnu.gidd.security.token.JwtToken;

public interface RefreshTokenService {

    RefreshToken saveRefreshToken(JwtToken token);

    RefreshToken getByJti(String jti);

    void invalidateSubsequentTokens(String jti);

    void rotateRefreshToken(JwtRefreshToken oldRefreshToken, JwtRefreshToken newRefreshToken);
}

File: IDATT2105 Full-stack applikasjonsutvikling/Rombestilling/backend/src/main/kotlin/ntnu/idatt2105/security/service/RefreshTokenService.kt
package ntnu.idatt2105.security.service

import ntnu.idatt2105.security.token.JwtRefreshToken
import ntnu.idatt2105.security.token.JwtToken
import ntnu.idatt2105.security.token.RefreshToken

interface RefreshTokenService {
    fun saveRefreshToken(token: JwtToken): RefreshToken

    fun getByJti(jti: String): RefreshToken

    fun invalidateSubsequentTokens(jti: String)

    fun rotateRefreshToken(oldRefreshToken: JwtRefreshToken, newRefreshToken: JwtRefreshToken)
}
